homeassistant/components/honeywell/climate.py

```diff
@@ -316,17 +320,20 @@ async def _turn_away_mode_on(self) -> None:
 
             # Set permanent hold
             # and Set temperature
-            away_temp = getattr(self, f"_{mode}_away_temp")
-            if mode == "cool":
+            if mode in COOLING_MODES:
                 self._device.set_hold_cool(True)
-                self._device.set_setpoint_cool(away_temp)
-            elif mode == "heat":
+                self._device.set_setpoint_cool(self._cool_away_temp)
+            elif mode in HEATING_MODES:
                 self._device.set_hold_heat(True)
-                self._device.set_setpoint_heat(away_temp)
+                self._device.set_setpoint_heat(self._heat_away_temp)
 
         except AIOSomecomfort.SomeComfortError:
+
             _LOGGER.error(
-                "Temperature %.1f out of range", getattr(self, f"_{mode}_away_temp")
+                "Mode %s Heat Temperature %.1f or Cool Temperature %.1f out of range",
```

Comment:
May be something like this reads out better? But I leave it up to you.
```suggestion
                "Temperature out of range. Mode: %s, Heat temperature: %.1f, Cool temperature: %.1f",
```

Reply:
Sure does.